tests/extmod/machine_timer.py

```diff
@@ -1,38 +1,53 @@
 # test machine.Timer
+import sys
 
 try:
-    import time, machine as machine
+    import time
+    import machine
 
     machine.Timer
 except:
     print("SKIP")
     raise SystemExit
 
+if "esp32" in sys.platform:
+    TIMER_ID = 0  # No virtual timer
+else:
+    TIMER_ID = -1
+
+    # Check the virtual timer ID argument is optional
+    t = machine.Timer(freq=5)
+    t.deinit()
+
 # create and deinit
-t = machine.Timer(freq=1)
+t = machine.Timer(TIMER_ID, freq=1)
 t.deinit()
 
 # deinit again
 t.deinit()
 
 # create 2 and deinit
-t = machine.Timer(freq=1)
-t2 = machine.Timer(freq=1)
+t = machine.Timer(TIMER_ID, freq=1)
+t2 = machine.Timer(TIMER_ID, freq=1)
```

Comment:
Since this is testing having multiple timers at once, probably this would need a separate `TIMER_ID2`.